### DISCUSSION TITLE
Static site generator creates additional "javascript" folders

### DISCUSSION THREAD

**Discussion by textplusdata** [datenstrom/community#629] [Report a bug] [answered]
Following up from discussion here https://github.com/datenstrom/yellow/discussions/617#discussioncomment-551450 below is a blog page with javascript using the experimental html parser extension. When creating a static suit by running build, a folder called "public javascript:toggle('toggledText1', 'togglerLink1');" is generated, both in the main yellow folder and as a sub-folder under "public". An index.html file is also generated in each folder.

I am not married to the javascript example itself, it is just an example. The same thing happens to script that is in the html body but calls a function from a js file loaded in the header.

Below is the page and then after that the code used for the html parser

```
---
Title: Html parser static website test
Published: 2021-03-09
Author: Text+Data
Layout: blog
Tag: Data visualisation
LayoutCustom: dataviz1
Parser: html
---
The script in this page results in a folder being made during static site generation.

<br>
<script type="text/javascript">
    // Show/Hide folded paragraph from
    // http://stackoverflow.com/questions/15719391/multiple-basic-slidetoggles-in-the-same-page-doesnt-work
            function toggle(id, link) {
                var ele = document.getElementById(id);
                var text = document.getElementById(link);
                if(ele.style.display === "block") {
                        ele.style.display = "none";
                    text.innerHTML = "show";
                }
                else {
                    ele.style.display = "block";
                    text.innerHTML = "hide";
                }
            };  
    // END SHOW/HIDE DIV	
</script> 
<br>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 

<br>
<br>
              
<a id="togglerLink1" href="javascript:toggle('toggledText1', 'togglerLink1');">hide</a> <span> &lt;== click here to show/hide the text below </span>
<div id="toggledText1" style="display:block">
<br>
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 
</div>      

<br>
<br>

Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

```

Code for html parser extension:

```
<?php
// Html extension, experimental

class YellowHtml {
   const VERSION = "0.8.17";

   // Handle page content in raw format
   public function onParseContentRaw($page, $text) {
       return $text;
   }
}

```
At the moment, I can resolve this by simply deleting the folders (am working on some php to do this automatically looking for any folders starting "public javascript" and deleting them).

**Comment by markseuffert** [comment]
I can't reproduce this on my computer, PHP 7.1.33 on a Macbook. I downloaded the latest version of Datenstrom Yellow, added your files and as far as I can see the static generator worked as expected. I also switched on debug mode and verified that no unnecessary folders were created. 

Which PHP version and operating system do you have? Do you have additional information for us?

**Reply by textplusdata** [reply]
Hi,
Thanks for checking!
Lubuntu 18.04.05
PHP 7.2.24-0ubuntu0.18.04.7 (cli)

I don't know if it makes a difference, but the blog is my first content folder.

I can try again with a totally vanilla install to double check it's not something I've changed somewhere.

**Reply by textplusdata** [reply]
Here is the detailed debug after fresh install of 8.17 version, with latest blog extension. This seems to be the problem: "YellowCommand::analyseLocations detected location:javascript:toggle('toggledText1', 'togglerLink1');/<br/>"

YellowSystem::load file:system/extensions/yellow-system.ini<br/>
YellowSystem::load Sitename:Datenstrom Yellow<br/>
YellowSystem::load Author:Datenstrom<br/>
YellowSystem::load Email:webmaster<br/>
YellowSystem::load Language:en<br/>
YellowSystem::load Layout:default<br/>
YellowSystem::load Theme:stockholm<br/>
YellowSystem::load Parser:markdown<br/>
YellowSystem::load Status:public<br/>
YellowSystem::load CoreStaticUrl:http://textplusdata.se<br/>
YellowSystem::load CoreServerUrl:auto<br/>
YellowSystem::load CoreServerTimezone:UTC<br/>
YellowSystem::load CoreMultiLanguageMode:0<br/>
YellowSystem::load CoreTrashTimeout:7776660<br/>
YellowSystem::load CoreMediaLocation:/media/<br/>
YellowSystem::load CoreDownloadLocation:/media/downloads/<br/>
YellowSystem::load CoreImageLocation:/media/images/<br/>
YellowSystem::load CoreExtensionLocation:/media/extensions/<br/>
YellowSystem::load CoreThemeLocation:/media/themes/<br/>
YellowSystem::load CoreMediaDirectory:media/<br/>
YellowSystem::load CoreDownloadDirectory:media/downloads/<br/>
YellowSystem::load CoreImageDirectory:media/images/<br/>
YellowSystem::load CoreSystemDirectory:system/<br/>
YellowSystem::load CoreExtensionDirectory:system/extensions/<br/>
YellowSystem::load CoreLayoutDirectory:system/layouts/<br/>
YellowSystem::load CoreThemeDirectory:system/themes/<br/>
YellowSystem::load CoreTrashDirectory:system/trash/<br/>
YellowSystem::load CoreCacheDirectory:cache/<br/>
YellowSystem::load CoreContentDirectory:content/<br/>
YellowSystem::load CoreContentRootDirectory:default/<br/>
YellowSystem::load CoreContentHomeDirectory:home/<br/>
YellowSystem::load CoreContentSharedDirectory:shared/<br/>
YellowSystem::load CoreContentDefaultFile:page.md<br/>
YellowSystem::load CoreContentErrorFile:page-error-(.*).md<br/>
YellowSystem::load CoreContentExtension:.md<br/>
YellowSystem::load CoreDownloadExtension:.download<br/>
YellowSystem::load CoreUserFile:yellow-user.ini<br/>
YellowSystem::load CoreLanguageFile:yellow-language.ini<br/>
YellowSystem::load CoreLogFile:yellow.log<br/>
YellowSystem::load UpdateExtensionUrl:https://github.com/datenstrom/yellow-extensions<br/>
YellowSystem::load UpdateExtensionFile:extension.ini<br/>
YellowSystem::load UpdateLatestFile:update-latest.ini<br/>
YellowSystem::load UpdateCurrentFile:update-current.ini<br/>
YellowSystem::load UpdateCurrentRelease:0.8.17<br/>
YellowSystem::load UpdateEventPending:none<br/>
YellowSystem::load UpdateEventDaily:1617321645<br/>
YellowSystem::load BlogNewLocation:@title<br/>
YellowSystem::load BlogPagesMax:5<br/>
YellowSystem::load BlogPaginationLimit:5<br/>
YellowSystem::load CommandStaticBuildDirectory:public/<br/>
YellowSystem::load CommandStaticDefaultFile:index.html<br/>
YellowSystem::load CommandStaticErrorFile:404.html<br/>
YellowSystem::load EditLocation:/edit/<br/>
YellowSystem::load EditUploadNewLocation:/media/@group/@filename<br/>
YellowSystem::load EditUploadExtensions:.gif, .jpg, .pdf, .png, .svg, .zip<br/>
YellowSystem::load EditKeyboardShortcuts:ctrl+b bold, ctrl+i italic, ctrl+k strikethrough, ctrl+e code, ctrl+s save, ctrl+alt+p preview<br/>
YellowSystem::load EditToolbarButtons:auto<br/>
YellowSystem::load EditEndOfLine:auto<br/>
YellowSystem::load EditNewFile:page-new-(.*).md<br/>
YellowSystem::load EditUserPasswordMinLength:8<br/>
YellowSystem::load EditUserHashAlgorithm:bcrypt<br/>
YellowSystem::load EditUserHashCost:10<br/>
YellowSystem::load EditUserHome:/<br/>
YellowSystem::load EditUserAccess:create, edit, delete, upload<br/>
YellowSystem::load EditLoginRestriction:0<br/>
YellowSystem::load EditLoginSessionTimeout:2592000<br/>
YellowSystem::load EditBruteForceProtection:25<br/>
YellowSystem::load ImageUploadWidthMax:1280<br/>
YellowSystem::load ImageUploadHeightMax:1280<br/>
YellowSystem::load ImageUploadJpgQuality:80<br/>
YellowSystem::load ImageThumbnailLocation:/media/thumbnails/<br/>
YellowSystem::load ImageThumbnailDirectory:media/thumbnails/<br/>
YellowSystem::load ImageThumbnailJpgQuality:80<br/>
YellowSystem::load MetaDefaultImage:favicon<br/>
YellowUser::load file:system/extensions/yellow-user.ini<br/>
YellowLanguage::load file:system/extensions/english.txt<br/>
YellowLanguage::load file:system/extensions/french.txt<br/>
YellowLanguage::load file:system/extensions/german.txt<br/>
YellowLanguage::load file:system/extensions/yellow-language.ini<br/>
YellowExtension::load file:system/extensions/blog.php<br/>
YellowExtension::load file:system/extensions/bundle.php<br/>
YellowExtension::load file:system/extensions/command.php<br/>
YellowExtension::load file:system/extensions/core.php<br/>
YellowExtension::load file:system/extensions/edit.php<br/>
YellowExtension::load file:system/extensions/english.php<br/>
YellowExtension::load file:system/extensions/french.php<br/>
YellowExtension::load file:system/extensions/german.php<br/>
YellowExtension::load file:system/extensions/html.php<br/>
YellowExtension::load file:system/extensions/image.php<br/>
YellowExtension::load file:system/extensions/markdown.php<br/>
YellowExtension::load file:system/extensions/meta.php<br/>
YellowExtension::load file:system/extensions/stockholm.php<br/>
YellowExtension::load file:system/extensions/update.php<br/>
YellowCore::getCommandInformation build<br/>
YellowContent::scanLocation location:<br/>
YellowContent::scanLocation location:root/<br/>
YellowLookup::findLocationFromFile / <- content/1-home/page.md<br/>
YellowLookup::findLocationFromFile /blog/ <- content/2-blog/page.md<br/>
YellowLookup::findLocationFromFile /shared/ <- content/shared/page.md<br/>
YellowContent::scanLocation location:/<br/>
YellowContent::scanLocation location:/blog/<br/>
YellowLookup::findLocationFromFile /blog/blog-example <- content/2-blog/2020-04-07-blog-example.md<br/>
YellowLookup::findLocationFromFile /blog/made-for-people <- content/2-blog/2020-12-06-made-for-people.md<br/>
YellowLookup::findLocationFromFile /blog/test <- content/2-blog/2020-12-06-test.md<br/>
YellowContent::scanLocation location:/shared/<br/>
YellowLookup::findLocationFromFile /shared/page-error-404 <- content/shared/page-error-404.md<br/>
YellowLookup::findLocationFromFile /shared/page-new-blog <- content/shared/page-new-blog.md<br/>
YellowLookup::findLocationFromFile /shared/page-new-default <- content/shared/page-new-default.md<br/>
Building static website 0%... YellowCommand::buildStaticFile status:200 location:/<br/>
Building static website 10%... YellowCommand::analyseLocations detected location:/blog/author:text+data/<br/>
YellowCommand::analyseLocations detected location:javascript:toggle('toggledText1', 'togglerLink1');/<br/>
YellowCommand::analyseLocations detected location:/blog/author:datenstrom/<br/>
YellowCommand::buildStaticFile status:200 location:/blog/<br/>
Building static website 20%... YellowCommand::analyseLocations detected location:/blog/tag:example/<br/>
YellowCommand::buildStaticFile status:200 location:/blog/blog-example<br/>
Building static website 35%... YellowCommand::buildStaticFile status:200 location:/blog/made-for-people<br/>
Building static website 45%... YellowCommand::analyseLocations detected location:/blog/tag:data-visualisation/<br/>
YellowCommand::buildStaticFile status:200 location:/blog/test<br/>
Building static website 60%... YellowCommand::buildStaticFile status:200 location:/blog/author:text+data/<br/>
Building static website 60%... YellowCommand::buildStaticFile status:200 location:javascript:toggle('toggledText1', 'togglerLink1');/<br/>
Building static website 60%... YellowCommand::buildStaticFile status:200 location:/blog/author:datenstrom/<br/>
Building static website 60%... YellowCommand::buildStaticFile status:200 location:/blog/tag:example/<br/>
Building static website 60%... YellowCommand::buildStaticFile status:200 location:/blog/tag:data-visualisation/<br/>
YellowCommand::buildStaticFile status:200 location:/media/downloads/yellow.pdf<br/>
YellowCommand::buildStaticFile status:200 location:/media/images/photo.jpg<br/>
YellowCommand::buildStaticFile status:200 location:/media/themes/stockholm-opensans-bold.woff<br/>
YellowCommand::buildStaticFile status:200 location:/media/themes/stockholm.png<br/>
YellowCommand::buildStaticFile status:200 location:/media/themes/stockholm.css<br/>
YellowCommand::buildStaticFile status:200 location:/media/themes/stockholm-opensans-regular.woff<br/>
YellowCommand::buildStaticFile status:200 location:/media/themes/stockholm-opensans-light.woff<br/>
YellowCommand::buildStaticFile status:200 location:/robots.txt<br/>
YellowCommand::buildStaticFile status:200 location:/publicjavascript:toggle('toggledText1', 'togglerLink1');/index.html<br/>
YellowCommand::buildStaticFile status:200 location:/error/<br/>
Building static website 100%... done
Yellow build: 20 files, 0 errors
YellowCore::command status:200 time:77 ms<br/>

**Reply by markseuffert** [reply]
Thanks for the detailed information. Now I know how to reproduce the problem.

**Comment by markseuffert** [comment, marked as answer]
Thanks for your bug report. I updated the [generate extension](https://github.com/annaesvensson/yellow-generate). When generating a static website it checks for content locations with arguments, for example pages with pagination. The existing code also detected hyperlinks with JavaScript and added them to content locations. 

Should be fixed now, please try it out.

**Reply by textplusdata** [reply]
I have tested and the fix seems to work just fine, thanks!
